Makefile

```diff
@@ -21,6 +21,7 @@ BUILDTAGS = btrfs_noversion libdm_no_deferred_remove $(PLATFORM_BUILD_TAG)
 BUILDFLAGS := -tags "$(BUILDTAGS)"
 
 PACKAGES := $(shell go list $(BUILDFLAGS) ./... | grep -v github.com/containers/image/vendor)
+PACKAGES_VET := $(shell echo $(PACKAGES) | grep -v github.com/containers/image/manifest)
```

Comment:
Yet again, no. Let’s actually find and implement a _real_ fix for the field inconsistency this highlights.

Reply:
Can we consider it a temporary patch? I am on go 1.12 already which means that I can't execute `go vet` on the code. As we don't know a "real fix" until now, I very much prefer being able to use the latest go.

Reply:
Things like that never stay temporary, and excluding a whole sub package tree is just too broad. I could maybe be persuaded to accept a patch that greps out a specific single error report (though that will be fragile longer term) — but I’d _much_ prefer a second pair of eyes independently reviewing and possibly testing the analysis https://github.com/containers/buildah/pull/1384 ; the possible fix arising from that would eventually end up also a trivial one-liner, but hopefully a correct one.

Reply:
There is no fix for it. It's valid json as the field of the embedded struct will just be shadowed. I did some research and found that this is a regression in go 1.12 and caused issues for many other users as well. It looks like it's being addressed soon (see https://github.com/golang/go/issues/30465).

I remove the commit although it's really annoying not being able to pass validation without manual editing.

Reply:
> It's valid json as the field of the embedded struct will just be shadowed.

“shadowed”? Actual JSON data of schema2 images in the wild either does not contain that field at all, or it (well, hopefully) contains data consistently in one of the two formats; a single JSON object, to my knowledge, just can’t contain two `parent` fields with different contents.

Whether or not Go becomes more lenient about this (and note that in the linked issue, the shadowed field has the same type, which is not our case), we still should have data structures that reflect how the data exists in the wild, and are unambiguous. The `parent` field name and the type can’t change because we can’t change JSON data in registries all over the world; our data structures definitely can change.

Reply:
My conclusion is that the duplicate name is intentional. Using the embedded struct avoids redundant fields while the JSON identifiers can be overwritten and only one `parent` will be written (i.e., the one of the main struct). As a `Digest` is a string, JSON doesn't seem to go south.